### PR TITLE
Fix the three System.Data.SqlClient tests that were failing on arm due to sni.dll not being available on arm

### PR DESCRIPTION
**Pull request by joperezr**
cc: @saurabh500 @danmosemsft @stephentoub 

Disabling few tests that are failing on our arm runs. Note that I couldn't find any logic in place to skip tests for ARM so I added some to PlatformDetection but please feel free to comment if there was already a way to do this.

### CODE REVIEW

File: src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ public void ConnectionTest()
 
 
         [PlatformSpecific(TestPlatforms.Windows)]  // Integ auth on Test server is supported on Windows right now
-        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))] // https://github.com/dotnet/corefx/issues/19218
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotArmProcessAndIsNotWindowsNanoServer))] // https://github.com/dotnet/corefx/issues/19218 And https://github.com/dotnet/corefx/issues/21598
```
Comment (danmosemsft) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs]: You should be able to do
` [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotArmProcess), nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))`

so you don't need to create IsNotArmProcessAndIsNotWindowsNanoServer

Comment (danmosemsft) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs]: I just notice it's params: `public ConditionalFactAttribute(params string[] conditionMemberNames)`

Comment (joperezr) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs]: Does that actually work, I thought that for conditional facts you needed to provide only one value

Comment (danmosemsft) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs]: didn't try but it should
https://github.com/dotnet/buildtools/blob/master/src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs#L41

**Pull request merged**